hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/bloom/HoodieBloomIndex.java

```diff
@@ -103,11 +102,6 @@ record -> new ImmutablePair<>(record.getPartitionPath(), record.getRecordKey()))
     // Step 3: Tag the incoming records, as inserts or updates, by joining with existing record keys
     HoodieData<HoodieRecord<R>> taggedRecords = tagLocationBacktoRecords(keyFilenamePairs, records, hoodieTable);
 
-    if (config.getBloomIndexUseCaching()) {
```

Comment:
I guess this was intentional. After this, taggedRecords is what is getting used. and we do cache that in BaseSparkCommitActionExecutor.execute
```
 @Override
  public HoodieWriteMetadata<HoodieData<WriteStatus>> execute(HoodieData<HoodieRecord<T>> inputRecords) {
    // Cache the tagged records, so we don't end up computing both
    JavaRDD<HoodieRecord<T>> inputRDD = HoodieJavaRDD.getJavaRDD(inputRecords);
    if (inputRDD.getStorageLevel() == StorageLevel.NONE()) {
      HoodieJavaRDD.of(inputRDD).persist(config.getTaggedRecordStorageLevel(),
          context, HoodieDataCacheKey.of(config.getBasePath(), instantTime));
    } else {
      LOG.info("RDD PreppedRecords was persisted at: " + inputRDD.getStorageLevel());
    }
.
.
```

So, not sure if we want to keep the persistance until the very end for these rdds which may not be used only.

Reply:
the main purpose of this PR is about fixing premature un-persisting like this example here.